src/target/riscv/riscv-013.c

```diff
@@ -1964,65 +1378,50 @@ static int write_memory(struct target *target, uint32_t address,
 				LOG_ERROR("Unsupported size: %d", size);
 				return ERROR_FAIL;
 		}
-		program_add32(program, addi(S0, S0, size));
+		program_add32(program, fence());
 		program_add32(program, ebreak());
-		write_program(target, program);
-		program_delete(program);
 
-		scans_t *scans = scans_new(target, count + 10);
-		for (uint32_t i = 0; i < count; i++) {
-			uint32_t value;
-			switch (size) {
-				case 1:
-					value = buffer[i];
-					break;
-				case 2:
-					value = buffer[2*i] | ((uint32_t) buffer[2*i+1] << 8);
-					break;
-				case 4:
-					value = buffer[4*i] |
-						((uint32_t) buffer[4*i+1] << 8) |
-						((uint32_t) buffer[4*i+2] << 16) |
-						((uint32_t) buffer[4*i+3] << 24);
-					break;
-				default:
-					return ERROR_FAIL;
-			}
-			scans_add_dmi_write(scans, DMI_DATA0, value, true);
-
-			if (i == 0) {
-				scans_add_dmi_write(scans, DMI_COMMAND,
-						AC_ACCESS_REGISTER_WRITE | AC_ACCESS_REGISTER_POSTEXEC
-						| abstract_register_size(xlen(target)) |
-						reg_number_to_no(S1), true);
-				scans_add_dmi_write(scans, DMI_ABSTRACTAUTO,
-						0x1 << DMI_ABSTRACTAUTO_AUTOEXECDATA_OFFSET,
-						false);
-			}
+		uint32_t value;
+		switch (size) {
+			case 1:
+				value = t_buffer[0];
+				break;
+			case 2:
+				value = t_buffer[0] | ((uint32_t) t_buffer[1] << 8);
+				break;
+			case 4:
+				value = t_buffer[0] |
+					((uint32_t) t_buffer[1] << 8) |
+					((uint32_t) t_buffer[2] << 16) |
+					((uint32_t) t_buffer[3] << 24);
+				break;
+			default:
+				return ERROR_FAIL;
 		}
-		int result = scans_execute(scans);
```

Comment:
I assume this is super slow now without the scans stuff? Is the plan to resurrect it at some point?